### PR TITLE
#94: eventuate-common-spring-reactive-jdbc contains regular spring-jd…

### PR DESCRIPTION
**Pull request by dartartem**
…bc dependencies (spring-boot-starter-jdbc). Extracted eventuate-common-reactive-jdbc from eventuate-common-spring-reactive-jdbc (necessary for eventuate-tram/eventuate-tram-core#143)

### CODE REVIEW

**Comment by kwonglau** [comment]
EventuateReactiveJdbcStatementExecutor#query should rename to EventuateReactiveJdbcStatementExecutor#queryForList

EventuateReactiveJdbcStatementExecutor#queryForList should rename to EventuateReactiveJdbcStatementExecutor#query

so that it matches the EventuateJdbcStatementExecutor interface

File: eventuate-common-spring-reactive-jdbc/src/main/java/io/eventuate/common/spring/jdbc/reactive/EventuateSpringReactiveJdbcStatementExecutor.java [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
+import io.eventuate.common.reactive.jdbc.EventuateReactiveRowMapper;
 import org.springframework.r2dbc.core.DatabaseClient;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
 import java.math.BigInteger;
 import java.util.Map;
 
-public class EventuateSpringReactiveJdbcStatementExecutor {
+public class EventuateSpringReactiveJdbcStatementExecutor implements EventuateReactiveJdbcStatementExecutor {
```
Comment (kwonglau) [inline eventuate-common-spring-reactive-jdbc/src/main/java/io/eventuate/common/spring/jdbc/reactive/EventuateSpringReactiveJdbcStatementExecutor.java:13]: in insertAndReturnId, we could use s.returnGeneratedValues to get the last insert id.


  public Mono<Long> insertAndReturnId(String sql,
                                      String idColumn,
                                      Object... params) {
    sql = reformatParameters(sql, params);
    DatabaseClient.GenericExecuteSpec genericExecuteSpec = bindParameters(databaseClient.sql(sql)
        .filter(s -> s.returnGeneratedValues(idColumn)), params);

    return genericExecuteSpec
        .map((row, rowMetadata) -> row.get(idColumn, Long.class))
        .first()
        .onErrorResume(t -> Optional.of(t)
            .filter(e -> e instanceof DataIntegrityViolationException)
            .<Mono<Long>>map(e -> Mono.error(new EventuateDuplicateKeyException(e)))
            .orElse(Mono.error(t)));
  }

Comment (cer) [reply eventuate-common-spring-reactive-jdbc/src/main/java/io/eventuate/common/spring/jdbc/reactive/EventuateSpringReactiveJdbcStatementExecutor.java:13]: @dartartem Please review this suggestion

Comment (dartartem) [reply eventuate-common-spring-reactive-jdbc/src/main/java/io/eventuate/common/spring/jdbc/reactive/EventuateSpringReactiveJdbcStatementExecutor.java:13]: will do

**Pull request merged**